pkg/skaffold/helm/bin.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2022 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package helm
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"regexp"
+
+	"github.com/blang/semver"
+	shell "github.com/kballard/go-shellquote"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/graph"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+)
+
+var (
+	// VersionRegex extracts version from "helm version --client", for instance: "2.14.0-rc.2"
+	VersionRegex = regexp.MustCompile(`v?(\d[\w.\-]+)`)
```

Comment:
nit: do we still have the same version constraints/checking-required moving from v1 -> v2?  For some reason I thought we maybe didn't need some checks (ok to keep as is though)

Reply:
I think we might still need version checking to differentiate between helm 3.0.x and helm 3.1+. I can do some investigation later and figure this out